addons/udes_stock/tests/test_picking.py

```diff
@@ -376,3 +375,33 @@ def test08_get_suggested_location_by_product_all_lots(self):
         # and the location with matching product but different lot.
         self.assertEqual(len(suggested_locations), 2)
         self.assertEqual(self.test_location_01 + self.test_location_02, suggested_locations)
+
+    def test09_considers_partially_available_move_lines_when_suggesting_location(self):
+        """Test that we don't suggest locations associated with partially
+        available move lines.
+        """
+        Location = self.env['stock.location']
+
+        drop_policy = "by_height_speed"
+        products_info = [{"product": self.apple, "qty": 10}]
+        self.product_category_slow = self.create_category(name='Slow')
+        self.product_category_ground = self.create_category(name='Ground')
+        self.apple.u_height_category_id = self.product_category_ground
+        self.apple.u_speed_category_id = self.product_category_slow
+
+        # We need another empty location
+        self.test_location_05 = Location.create({
+            "name": "Test location 05",
+            "barcode": "LTEST05",
+            "location_id": self.stock_location.id,
+        })
+
+        picking1 = self.create_and_assign_putaway_picking(products_info, drop_policy)
+        picking2 = self.create_and_assign_putaway_picking(products_info, drop_policy)
+        picking1.apply_drop_location_policy()
+        picking1.move_lines[0].product_uom_qty += 1
```

Comment:
Shall we assert that the state is now partially available ?

Reply:
asserted